### PR TITLE
App mobile responsive (#35)

### LINKED ISSUE #35: Mobile / small screen view
Currently, the calendar is not mobile friendly. Since the calendar is also embedded in a larger webpage, it might be best to focus the effort to viewport size instead of device type.

### PR DESCRIPTION
**Pull request by gautamjuyal**
## Fixes #35 

Adds mobile responsive view for the app

## What does this PR do?

- Adds proper icons to the event details popup
- Adds event location field to the event details popup
- Places the event date right below the title with a slight smaller font
- For desktop view, creates event details popup near the event clicked (inspired from google calendar)
- Increases z-index of event-details popup (previously it would hide under 'more events' menu)
- Sets aspect ratio of calendar based on viewport width
- Sets calendar view to weekly by default for mobile view
- Sets event details popup to full screen for mobile view

## Screenshots

1. Popup position next to event clicked
Screenshot (359)

2. Mobile default view for calendar
Screenshot (360)

3. Mobile view for calendar month view
Screenshot (361)

4. Event datails screen for mobile
Screenshot (362)

## New dependencies added

1. [mdi-react](https://www.npmjs.com/package/@mdi/react) : Provides `<Icon>` component
2. [mdi-js](https://www.npmjs.com/package/@mdi/js) : Provides icons for the `<Icon>` component

Icons provided by [Material Design Icons](https://pictogrammers.com/library/mdi/)

### CODE REVIEW

**Review by maoo — APPROVED** [review]
LGTM! I'm not sure if there's a way to simplify the popup positioning block, but since it seems to work, we can also improve in a second stage.

@TheJuanAndOnly99 - can you please review?

Thanks!

File: src/App.jsx [hunk 1]
```diff
@@ -22,10 +24,28 @@ function App() {
 
 	useEscKey(() => setShowEventDetails(false));
 
+	const [popupPosition, setPopupPosition] = useState({})
+
+	const createPopupPosition = (event)=>{
```
Comment (mimiflynn) [inline src/App.jsx:29]: This should be wrapped in a `useMemo`

https://react.dev/reference/react/useMemo#usememo

File: src/App.jsx [hunk 2]
```diff
@@ -132,7 +161,9 @@ function App() {
 			<FullCalendar
 				ref={calendarRef}
 				plugins={[ dayGridPlugin, iCalendarPlugin, interactionPlugin, timeGridPlugin, rrulePlugin ]}
-				initialView="dayGridMonth"
+				initialView={window.outerWidth > 600 ? "dayGridMonth" : "timeGridWeek"}
```
Comment (mimiflynn) [inline src/App.jsx:164]: This kind of logic (line 164 and 166) should be in a `useState` hook.

**Review by mimiflynn — CHANGES_REQUESTED** [review]
Thanks for digging into this!

File: src/App.jsx (hunk 1 above)
Comment (gautamjuyal) [reply src/App.jsx:29]: Is that possible since the function takes `event` as argument?

Comment (gautamjuyal) [reply src/App.jsx:29]: is it possible since the function takes `event` as argument?

Comment (mimiflynn) [reply src/App.jsx:29]: Yes, sorry, you're right. A refactor would only catch the use case of clicking the same event twice, which is unnecessary.

Comment (gautamjuyal) [reply src/App.jsx:29]: That's totally cool. I tried to modify the function to fit useMemo in, but it complicated the things even more. Let's keep it for now and refactor it in future maybe.

**Review by mimiflynn — APPROVED** [review]
Thanks for making those updates!

**Comment by TheJuanAndOnly99** [comment]
Thanks @gautamjuyal! Looks great :)

**Comment by gautamjuyal** [comment]
Glad to contribute :)

**Pull request merged**